modules/system/classes/PluginBase.php

```diff
@@ -321,4 +328,104 @@ protected function getConfigurationFromYaml($exceptionMessage = null)
 
         return $this->loadedYamlConfiguration;
     }
+
+    /**
+     * Returns a list of plugins replaced by this plugin
+     *
+     * @return array|string[]
+     */
+    public function getReplaces(): array
+    {
+        $replaces = $this->pluginDetails()['replaces'] ?? null;
+
+        if (is_array($replaces)) {
+            return array_keys($replaces);
+        }
+
+        return is_string($replaces) ? [$replaces] : [];
+    }
+
+    /**
+     * Returns bool for if a plugin is replaced
+     * @param string $pluginIdentifier
+     * @param string $version
+     * @return bool
+     */
+    public function replaces(string $pluginIdentifier, string $version): bool
+    {
+        $replaces = $this->pluginDetails()['replaces'] ?? null;
+
+        if (is_string($replaces) && $replaces === $pluginIdentifier) {
+            return true;
+        }
+
+        if (is_array($replaces) && in_array($pluginIdentifier, array_keys($replaces))) {
+            return $this->replacesVersion($version, $replaces[$pluginIdentifier]);
+        }
+
+        return false;
+    }
+
+    /**
+     * Passes the check through to composer/semver but allows Winter to perform
+     * checks / changes as required
+     *
+     * @return bool
+     */
+    protected function replacesVersion(string $version, string $constraints): bool
+    {
+        if ($constraints === 'self.version') {
+            $constraints = $this->getVersion();
```

Comment:
Shouldn't `self.version` refer to the last advertised version available in version.yaml rather than the current version of the plugin present in the migrations table?